src/STEMMUS_SCOPE.m

```diff
@@ -815,7 +797,38 @@
             hSAVE = hN;
         end
     end
-    run Forcing_PARM;
+    %% Reconstruct the driving forces to be a continuous function of time
```

Comment:
I saw that earlier it had the following comment
```
       Reconstruct the driving forces e.g. Ta, RH, Wind, Pressure,
        precipitation, etc. to be a continuous function of time via Fourier
        transformation, e.g. through identifying the dominant frequency etc.
```
Which makes me feel like this is part of never-finished code, which might as well be fully removed.

Reply:
> I saw that earlier it had the following comment
> 
> ```
>        Reconstruct the driving forces e.g. Ta, RH, Wind, Pressure,
>         precipitation, etc. to be a continuous function of time via Fourier
>         transformation, e.g. through identifying the dominant frequency etc.
> ```
> 
> Which makes me feel like this is part of never-finished code, which might as well be fully removed.

Hi @BSchilperoort. The Fourier transformation was used to transfer the 'discrete' (e.g., half-houly or hourly) meto forcing to a continuous one (e.g., at second interval). This is done as such, because original STEMMUS model use the adaptive time intervals (from 1s to 1800s). This adaptive time interval is needed for making the model converge during precipitation events. 

During a rainfall event, soil moisture will change dramatically, then the numerical solution will not converge. However, if we reduce the time step (for the numerical solution) from 1800s to 900s, and in case not yet converged, reduced it further to 450s, and so on (till 1s as the shortest time step for the numerical solution). 

As such, we need to have the corresponding meto-forcing data for the above corresponding time steps (i.e. 900s, 450s etc.).

One can argue that we can just use the same meto-forcing, even the time-step is reduced. In that sense, we can indeed delete this part. Since, currently, we are not using the adaptive time steps in STEMMUS-SCOPE.

Reply:
> Since, currently, we are not using the adaptive time steps in STEMMUS-SCOPE.

Do you mean that `Forcing_PARM.m` implements the "adaptive time steps" and currently the `Forcing_PARM.m` is not used in the main script `STEMMUS_SCOPE.m`? note that the main script runs `Forcing_PARM.m` at line [818](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/51668dcbc747ecc02f34473476a80412baa08bdd/src/STEMMUS_SCOPE.m#L818). I found that variables like `Rns` and `Rn` are calculated in `Forcing_PARM.m` script but never been used by other scripts. But `Forcing_PARM.m` also includes calculations like `HR_a(KT) = 0.01 .* (RH_msr(KT));` and `HR_a` is used in `Eval_Cal.m` script. Or `TopPg(KT) = 100 .* (Pg_msr(KT));` and  `TopPg` is used in `Air_BS.m`. Can you please explain these codes? are these related to "the adaptive time steps" and should be removed or these are only unit conversions and should be kept?

Reply:
Hi Sarah, i am sorry for causing confusion. When i responded to Bart, i referred to earlier comment cited as "Reconstruct the driving forces e.g. Ta, ...". What i meant 'we can indeed remove this part', is only referring to the part linked to Fourier transformation.

And the Fourier transformation part has been removed.

Reply:
> note that the main script runs `Forcing_PARM.m` at line [818](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/51668dcbc747ecc02f34473476a80412baa08bdd/src/STEMMUS_SCOPE.m#L818). I found that variables like `Rns` and `Rn` are calculated in `Forcing_PARM.m` script and used by other scripts. But `Forcing_PARM.m` also includes calculations like `HR_a(KT) = 0.01 .* (RH_msr(KT));` and `HR_a` is used in `Eval_Cal.m` script. Or `TopPg(KT) = 100 .* (Pg_msr(KT));` and `TopPg` is used in `Air_BS.m`. Can you please explain these codes? are these related to "the adaptive time steps" and should be removed or should be kept?

Yes, Sarah, these are needed in each time step to get the driving force to drive the STEMMUS-SCOPE. Those lines in 'Forcing_PARM' are used to convert unit to Centimeter-Gram-Second system.

They are not related to 'adaptive time steps'.